### DISCUSSION TITLE
How to use OpenAI's O1-Preview model ？

### DISCUSSION THREAD

**Discussion by lam584** [danny-avila/LibreChat#4037] [Q&A] [answered]
I added the o1-preview model to the .env file, but I got an error when using it:
![image](https://github.com/user-attachments/assets/2e7723da-ec59-4540-9e79-7061dc32fdb8)
![image](https://github.com/user-attachments/assets/5badbf9d-72eb-4d7c-ac6a-a2c6f4dbc108)

**Comment by danny-avila** [comment]
https://x.com/LibreChatAI/status/1834560654285373520

**Reply by lam584** [reply]
I made sure that LibreChat v0.7.5 was installed correctly in docker and that the other models worked fine as well, but I couldn't use the O1-Preview model.

**Reply by danny-avila** [reply]
![image](https://github.com/user-attachments/assets/3b33b42e-e023-4335-b3c2-a0cdbbcb25b5)

It should work if you follow my instructions but you also need to be API tier 5:
https://platform.openai.com/settings/organization/limits

And if you do:
https://community.openai.com/t/were-tier-5-usage-but-dont-have-access-to-o1-models/937904/4
"""
OpenAI confirmed that we should have access this afternoon, Pacific time.
"""

**Comment by fbaeumer** [comment]
Same problem for us. We have Tier-5 but in OpenAI the o1-mini model is listed and the others are listed as text models. Is that the problem? We get the same error message.

```
OPENAI_MODELS=gpt-3.5-turbo,gpt-4,gpt-4-turbo,gpt-4o,gpt-4o-mini,o1-preview,o1-mini,o1-mini-2024-09-12
DEBUG_OPENAI=true
# TITLE_CONVO=false
# OPENAI_TITLE_MODEL=gpt-3.5-turbo
OPENAI_SUMMARIZE=true
OPENAI_SUMMARY_MODEL=gpt-3.5-turbo
# OPENAI_FORCE_PROMPT=true
# OPENAI_REVERSE_PROXY=
```

<img width="749" alt="image" src="https://github.com/user-attachments/assets/5ea8931c-b941-4258-a746-6947d8ccf6d4">

Version is Librechat 7.5.-rc2

**Reply by SharpLu** [reply]
Hi @danny-avila , We have Tier-5 as well, when try librechat docker image we got the same issue, any suggestions? Thank you so much
<img width="1093" alt="Screenshot 2024-09-17 at 13 16 38" src="https://github.com/user-attachments/assets/7f5efbf0-60a3-4be2-8dc9-79fa0a355d39">

**Reply by SharpLu** [reply]
we tried the o1-preview, o1-mini, o1-preview-2024-09-12,o1-mini-2024-09-12 all had the same error.

**Reply by danny-avila** [reply]
You need to update to the latest dev build to use it:

![image](https://github.com/user-attachments/assets/dd113ad6-70dd-494c-a363-579e99262978)

**Reply by fbaeumer** [reply]
image: ghcr.io/danny-avila/librechat-dev:latest

Should be the correct version in our case. @SharpLu if you find a way, would be thankful to hear your solution.

**Reply by danny-avila** [reply]
Please review my comment on making sure you are using the latest updates:

https://github.com/danny-avila/LibreChat/issues/2620#issuecomment-2339357018

**Reply by SharpLu** [reply]
Hi @fbaeumer  Yes , it worked for me for LibreChat v0.7.5-rc2, 
i believe you have to delete the old repo and do it again follow https://www.librechat.ai/docs/local/docker
**don't forget to update your env variables.**

**Reply by fbaeumer** [reply]
Can you further explain the "don't forget to update your env variables." part?
I already tried all the tricks Danny mentioned in the other thread.

**Reply by SharpLu** [reply]
I have included those parameters inside the .env file under docker image librechat folder.
OPENAI_API_KEY=xxxxx
OPENAI_MODELS=gpt-4o,chatgpt-4o-latest,o1-preview,o1-preview,o1-mini,o1-mini-2024-09-12,o1-preview-2024-09-12

**Reply by fbaeumer** [reply]
Yes, thank you. I have of course set these. The models are also displayed. 
I think I'm making a stupid mistake somewhere. I'll just wait until it arrives in another version, if necessary....

**Reply by Odrec** [reply]
We're having the same issue.

**Comment by SharpLu** [comment]
Hi @danny-avila again,
I have tried to use the chatgpt o1-preview and o1-mini, i got the error 

Something went wrong. Here's the specific error message we encountered: Failed to send message. HTTP 404 - { "error": { "message": "This is a chat model and not supported in the v1/completions endpoint. Did you mean to use v1/chat/completions?", "type": "invalid_request_error", "param": "model", "code": null } }

a.) The version i'm used [LibreChat v0.7.5-rc2] helm chart deployed on kubernates
b.)The deployment worked with chatgpt 4o and other models.
c.)the OPENAI_API_KEY are configured on my environment 

---
But it worked for the docker image on my local, any suggestions? or is a bug we have to wait for new releases? Thank you 
![Screenshot 2024-10-02 at 11 10 49](https://github.com/user-attachments/assets/07ebe517-db07-4889-a469-bef6ac7ac4a9)

**Comment by jmrobles** [comment]
I guess the problem is that the API uses a old version of "node-openai". This version does not include the new models, so it tries to use the completion endpoints instead of the chat completion endpoint. 

v4.47.1 => https://github.com/openai/openai-node/blob/d7d5610d91573740d7e3c27e4afde5650ee6e349/src/resources/chat/chat.ts

latest =>https://github.com/openai/openai-node/blob/master/src/resources/chat/chat.ts

**Reply by Odrec** [reply]
If this is the issue, how come the latest librechat version is still using the old openai API and why do the new models work for some people and not for others?

**Reply by danny-avila** [reply]
Not everyone has access, that could also be the issue.

I would like to cut a new release today, which should make o1 more available to those not on the dev build.

**Reply by Odrec** [reply]
Our organisation is tier 5 and I can use the o1 models on the Playground or with curl so we have access. But we get this error when trying them on librechat. But that's good to know that you're doing a new release! Hopefully that solves the issue.

**Reply by Odrec** [reply]
So one person in our team found the issue. We were doing this in the docker container:

```image: ghcr.io/danny-avila/librechat:{{librechat_version}}```

Which loaded the 0.75-rc2 version but not the dev one apparently. So when they changed it to the dev branch:

```image: ghcr.io/danny-avila/librechat-dev:latest```

It finally worked. Hopefully this helps someone having a similar problem.

**Comment by danny-avila** [comment, marked as answer]
v0.7.5 is now live which includes the o1 updates. If you are seeing a green icon and an error, you're using an older version of LibreChat.

At the time of writing, the docker image for releases is still building, but should be done momentarily.

https://www.librechat.ai/changelog/v0.7.5

**Comment by krzemian** [comment]
v0.7.6, I'm on Tier 1 but have access to o1-preview according to https://platform.openai.com/settings/organization/limits
I've set the API token
However, the o1 models are not on the list. How do I troubleshoot it further?

Tested access on Witsy, all works

**Reply by lam584** [reply]
You can specify the desired models by configuring the OPENAI_MODELS environment variable. For instance:
OPENAI_MODELS=o1-preview,o1-preview-2024-09-12,o1-mini,o1-mini-2024-09-12,gpt-4o,chatgpt-4o-latest,gpt-4o-mini,Whisper,TTS,TTS HD,gpt-3.5-turbo-0125,gpt-3.5-turbo-0301,gpt-3.5-turbo,gpt-4,gpt-4-0613,gpt-4-vision-preview,gpt-3.5-turbo-0613,gpt-3.5-turbo-16k-0613,gpt-4-0125-preview,gpt-4-turbo-preview,gpt-4-1106-preview,gpt-3.5-turbo-1106,gpt-3.5-turbo-instruct,gpt-3.5-turbo-instruct-0914,gpt-3.5-turbo-16k
You can find an .env file in the root directory of LibreChat, which you can edit to modify environment variables

**Comment by copley** [comment]
I have the same issue :(

Are you able to tell me why this code returns these errors?, 
An error occurred with model o1: Completions.create() got an unexpected keyword argument 'max_completion_tokens'
An error occurred with model o3-mini: Completions.create() got an unexpected keyword argument 'max_completion_tokens'
An error occurred with model o1-pro: Error code: 404 - {'error': {'message': 'This is not a chat model and thus not supported in the v1/chat/completions endpoint. Did you mean to use v1/completions?', 'type': 'invalid_request_error', 'param': 'model', 'code': None}}
An error occurred with model o1-mini: Completions.create() got an unexpected keyword argument 'max_completion_tokens'


Please see my code below, 

import os
from dotenv import load_dotenv
from openai import OpenAI

# Load environment variables from .env file
load_dotenv()

# Retrieve the API key from environment variables
api_key = os.getenv("OPENAI_API_KEY")
if api_key is None:
    raise ValueError("API key not found. Please set the OPENAI_API_KEY environment variable.")

# Initialize OpenAI client
client = OpenAI(api_key=api_key)

# List of models to test
models = [
    "gpt-4o",
    "gpt-3.5-turbo-instruct",
    "davinci-002",
    "babbage-002",
    "o1",
    "o3-mini",
    "o1-pro",
    "o1-mini"
]

# Prompt to send to each model
prompt = "What is the capital of France?"

# Iterate over each model and get a response
for model in models:
    try:
        if model in ["o1", "o3-mini", "o1-mini"]:
            # Use the completions endpoint with max_completion_tokens for o1 series models
            response = client.completions.create(
                model=model,
                prompt=prompt,
                max_completion_tokens=50
            )
            answer = response.choices[0].text.strip()
        elif model in ["gpt-3.5-turbo-instruct"]:
            # Use the completions endpoint with max_tokens for gpt-3.5-turbo-instruct
            response = client.completions.create(
                model=model,
                prompt=prompt,
                max_tokens=50
            )
            answer = response.choices[0].text.strip()
        elif model in ["davinci-002", "babbage-002"]:
            # Use the completions endpoint with max_tokens for davinci-002 and babbage-002
            response = client.completions.create(
                model=model,
                prompt=prompt,
                max_tokens=50
            )
            answer = response.choices[0].text.strip()
        else:
            # Use the chat completions endpoint with max_tokens for chat models
            response = client.chat.completions.create(
                model=model,
                messages=[{"role": "user", "content": prompt}],
                max_tokens=50
            )
            answer = response.choices[0].message.content.strip()

        print(f"Response from {model}: {answer}")

    except Exception as e:
        print(f"An error occurred with model {model}: {str(e)}")






 
Response from gpt-4o: The capital of France is Paris.
Response from gpt-3.5-turbo-instruct: The capital of France is Paris.
Response from davinci-002: What is the capital of Italy? What is the capital of England? Meikle suggests that classrooms should head outside to ask these questions aloud. Learning about geography can also consist of students creating a map, whether that be drawing a map of the classroom
Response from babbage-002: Which is the capital?

Paris, capital of France.

In India, Mumbai is the capital. Capital.

Capital of France is Paris.

Capital of France is a capital?

I think so.

Yes. Okay. Capital of India is Bangalore. Capital of
An error occurred with model o1: Completions.create() got an unexpected keyword argument 'max_completion_tokens'
An error occurred with model o3-mini: Completions.create() got an unexpected keyword argument 'max_completion_tokens'
An error occurred with model o1-pro: Error code: 404 - {'error': {'message': 'This is not a chat model and thus not supported in the v1/chat/completions endpoint. Did you mean to use v1/completions?', 'type': 'invalid_request_error', 'param': 'model', 'code': None}}
An error occurred with model o1-mini: Completions.create() got an unexpected keyword argument 'max_completion_tokens'

**Reply by danny-avila** [reply]
Update to latest.

Updating Instructions (docker):
```bash
# Linux command to Remove all existing images
docker images -a | grep "librechat" | awk '{print $3}' | xargs docker rmi

# Windows Powershell
docker images -a | findstr "librechat" | ForEach-Object { docker rmi $_.Split()[2] }
```

Then follow the directions here: https://www.librechat.ai/docs/local/docker#update-librechat
